src/mz/src/configuration.rs

```diff
@@ -0,0 +1,277 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::{
+    collections::{BTreeMap, HashMap},
+    fmt::Display,
+    fs,
+    path::PathBuf,
+};
+
+use anyhow::{bail, Context, Ok, Result};
+use dirs::home_dir;
+use reqwest::header::{HeaderMap, HeaderValue, CONTENT_TYPE, USER_AGENT};
+use reqwest::Client;
+use serde::{Deserialize, Serialize};
+use uuid::Uuid;
+
+use crate::utils::CloudProviderRegion;
+
+#[derive(Serialize, Deserialize, Debug)]
+struct Profile0 {
+    email: String,
+    #[serde(rename(serialize = "app-password", deserialize = "app-password"))]
+    app_password: String,
+    region: Option<CloudProviderRegion>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub(crate) struct Configuration {
+    #[serde(skip)]
+    modified: bool,
+    default_profile: String,
```

Comment:
The field `default_profile` also appears in the config file `config.toml` resulting in two _defaults_. Is it intentional?

E.g.:
```
default_profile = 'www'
[profiles.default]
```

Reply:
I'll rename to current_profile. That's what it really is.